chat-plugins/uno.js

```diff
@@ -0,0 +1,576 @@
+
+const rgb_Gradients = {
+	"Green": "rgba(0, 122, 0, 1), rgba(0, 185, 0, 0.9)",
+	"Yellow": "rgba(255, 225, 0, 1), rgba(255, 255, 85, 0.9)",
+	"Blue": "rgba(40, 40, 255, 1), rgba(125, 125, 255, 0.9)",
+	"Red": "rgba(255, 0, 0, 1), rgba(255, 125, 125, 0.9)",
+	"Black": "rgba(0, 0, 0, 1), rgba(0, 0, 0, 0.55)",
+};
+
+const text_shadow = "text-shadow: 1px 0px black, -1px 0px black, 0px -1px black, 0px 1px black, 2px -2px black;";
+
+function cardHTML(card, fullsize) {
+	let surface = card.value.replace(/[^A-Z0-9\+]/g, "");
+	let background = rgb_Gradients[card.colour];
+	if (surface === "R") surface = '<i class="fa fa-refresh" aria-hidden="true"></i>';
+
+	return `<button class="button" style="font-size: 14px; font-weight: bold; color: white; ${text_shadow} padding-bottom: 117px; text-align: left; height: 135px; width: ${fullsize ? "72" : "37"}px; border-radius: 10px 2px 2px 3px; color: white; background: ${card.colour}; background: -webkit-linear-gradient(${background}); background: -o-linear-gradient(${background}); background: -moz-linear-gradient(${background}); background: linear-gradient(${background});" name=send value="/uno play ${card.name}">${surface}</button>`;
+}
+
+function createDeck() {
+	let colours = ["Red", "Blue", "Green", "Yellow"];
+	let values = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "Reverse", "Skip", "+2"];
+
+	let basic = [];
+
+	for (let i = 0; i < 4; i++) {
+		let colour = colours[i];
+		basic.push(...values.map(v => {
+			return {value: v, colour: colour, name: colour + " " + v};
+		}));
+	}
+
+	return [...basic, ...basic, // two copies of the basic stuff (total 96)
+		...[0, 1, 2, 3].map(v => ({colour: colours[v], value: "0", name: colours[v] + " 0"})), // the 4 0s
+		...[0, 1, 2, 3].map(v => ({colour: "Black", value: "Wild", name: "Wild"})), // wild cards
+		...[0, 1, 2, 3].map(v => ({colour: "Black", value: "+4", name: "Wild +4"})), // wild +4 cards
+	]; // 108 cards
+}
+
+class UNOgame extends Rooms.RoomGame {
+	constructor(room, cap) {
+		super(room);
+
+		if (room.gameNumber) {
+			room.gameNumber++;
+		} else {
+			room.gameNumber = 1;
+		}
+
+		this.allowRenames = true;
+		this.playerCap = parseInt(cap) || 6;
+		this.maxTime = maxTime;
+
+		this.gameid = "uno";
+		this.title = "UNO";
+
+		this.state = "signups";
+
+		this.currentPlayer = null;
+		this.deck = Tools.shuffle(createDeck());
+		this.discards = [];
+		this.topCard = null;
+
+		this.direction = 1;
+	}
+
+	onStart() {
+		if (this.playerCount < 2) return false;
+		this.sendToRoom(`|uhtmlchange|uno-${this.room.gameNumber}|<div class="infobox">The game of UNO has started.</div>`);
+		this.state = "play";
+
+		this.onNextPlayer();  // determines the first player
+
+		// give cards to the players
+		for (let i in this.players) {
+			this.players[i].hand.push(...this.drawCard(7));
+		}
+
+		// top card of the deck.
+		do {
+			this.topCard = this.drawCard(1)[0];
+			this.discards.unshift(this.topCard);
+		} while (this.topCard.colour === "Black");
```

Comment:
Um, maybe use `.color` since afaik the rest of the codebase is in US English? I won't insist though.